bukuserver/views.py

```diff
@@ -85,6 +85,13 @@ def _filter_arg(flt):
         """Exposes filter slugify logic; works because BookmarkModelView.named_filter_urls = True"""
         return BaseModelView.get_filter_arg(BookmarkModelView, None, flt)
 
+    def _saved(self, id, url, ok):
+        if ok:
+            session['saved'] = id
+        else:
+            raise Exception('Duplicate URL' if self.model.bukudb.get_rec_id(url) not in [-1, id] else
```

Comment:
why not use flask abort here https://flask.palletsprojects.com/en/2.2.x/api/#flask.abort?

Reply:
Because this code isn't generating a failure response, it's producing an error message – the exception is handled in the same place where the method is invoked (see lines 209 & 420).

When running in production mode, this results in displaying an error message to the user, above the form containing entered data.
![image](https://user-images.githubusercontent.com/16271003/208443852-27750178-e1e4-4d83-8fae-203ff80fd41c.png)
![image](https://user-images.githubusercontent.com/16271003/208444667-22d36f7b-ccea-4ea0-8047-737d14644324.png)